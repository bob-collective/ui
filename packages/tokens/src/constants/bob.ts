import { ChainId } from '@gobob/chains';

import { TBTC, USDC, USDT, WBTC } from './common';
import { WETH9 } from './weth';

export const bobTokens = {
  weth: WETH9[ChainId.BOB],
  wbtc: WBTC[ChainId.BOB],
  tbtc: TBTC[ChainId.BOB],
  usdc: USDC[ChainId.BOB],
  usdt: USDT[ChainId.BOB]
};

export const oldBobSepoliaTokens = {
  weth: WETH9[ChainId.OLD_BOB_SEPOLIA],
  wbtc: WBTC[ChainId.OLD_BOB_SEPOLIA],
  tbtc: TBTC[ChainId.OLD_BOB_SEPOLIA],
  usdc: USDC[ChainId.OLD_BOB_SEPOLIA],
  usdt: USDT[ChainId.OLD_BOB_SEPOLIA]
};
