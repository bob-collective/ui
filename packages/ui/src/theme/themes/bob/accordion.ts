import { fontWeight, rounded, spacing, typography } from '../../core';
import { AccordionTheme } from '../../components';

import { color } from './colors';

const accordion: AccordionTheme = {
  base: {
    borderRadius: rounded('md'),
    border: `1px solid ${color('grey-300')}`
  },
  item: {
    base: {
      '&:not(:last-of-type)': {
        borderBottom: `1px solid ${color('grey-300')}`
      }
    },
    button: {
      padding: spacing('xl')
    },
    heading: {
      ...typography('lg'),
      fontWeight: fontWeight('medium')
    },
    content: {
      paddingTop: spacing('none'),
      paddingLeft: spacing('xl'),
      paddingRight: spacing('xl'),
      paddingBottom: spacing('xl')
    }
  }
};

export { accordion };
