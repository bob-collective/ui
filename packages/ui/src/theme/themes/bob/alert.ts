import { rounded, spacing, fontSize, typography } from '../../core';
import { AlertTheme } from '../../components';

import { color } from './colors';

const alert: AlertTheme = {
  base: {
    ...typography('s'),
    color: color('dark'),
    padding: `${spacing('md')} ${spacing('xl')}`,
    borderRadius: rounded('md')
  },
  icon: {
    padding: `${spacing('s')} 0`,
    marginRight: spacing('md')
  },
  title: {
    fontSize: fontSize('md', 'rem'),
    marginBottom: spacing('s')
  },
  content: {
    padding: `${spacing('s')} 0`
  },
  closeBtn: {
    width: spacing('4xl'),
    height: spacing('4xl')
  },
  closeBtnWrapper: {
    marginRight: `-${spacing('md')}`
  },
  status: {
    success: {
      filled: {
        base: {
          backgroundColor: color('green-400'),
          color: color('dark')
        },
        icon: {
          color: color('dark')
        },
        title: {
          color: color('dark')
        }
      },
      outlined: {
        base: {
          backgroundColor: color('green-900'),
          border: `1px solid ${color('green-400')}`,
          color: color('light')
        },
        icon: {
          color: color('green-400')
        },
        title: {
          color: color('green-400')
        }
      },
      default: {
        base: {
          backgroundColor: color('green-900'),
          color: color('light')
        },
        icon: {
          color: color('green-400')
        },
        title: {
          color: color('green-400')
        }
      }
    },
    error: {
      filled: {
        base: {
          backgroundColor: color('red-600'),
          color: color('dark')
        },
        icon: {
          color: color('dark')
        },
        title: {
          color: color('dark')
        }
      },
      outlined: {
        base: {
          backgroundColor: color('red-900'),
          border: `1px solid ${color('red-600')}`,
          color: color('light')
        },
        icon: {
          color: color('red-600')
        },
        title: {
          color: color('red-600')
        }
      },
      default: {
        base: {
          backgroundColor: color('red-900'),
          color: color('light')
        },
        icon: {
          color: color('red-600')
        },
        title: {
          color: color('red-600')
        }
      }
    },
    info: {
      filled: {
        base: {
          backgroundColor: color('blue-400'),
          color: color('dark')
        },
        icon: {
          color: color('dark')
        },
        title: {
          color: color('dark')
        }
      },
      outlined: {
        base: {
          backgroundColor: color('blue-900'),
          border: `1px solid ${color('blue-400')}`,
          color: color('light')
        },
        icon: {
          color: color('blue-400')
        },
        title: {
          color: color('blue-400')
        }
      },
      default: {
        base: {
          backgroundColor: color('blue-900'),
          color: color('light')
        },
        icon: {
          color: color('blue-400')
        },
        title: {
          color: color('blue-400')
        }
      }
    },
    warning: {
      filled: {
        base: {
          backgroundColor: color('yellow-500'),
          color: color('dark')
        },
        icon: {
          color: color('dark')
        },
        title: {
          color: color('dark')
        }
      },
      outlined: {
        base: {
          backgroundColor: color('yellow-900'),
          border: `1px solid ${color('yellow-500')}`,
          color: color('light')
        },
        icon: {
          color: color('yellow-500')
        },
        title: {
          color: color('yellow-500')
        }
      },
      default: {
        base: {
          backgroundColor: color('yellow-900'),
          color: color('light')
        },
        icon: {
          color: color('yellow-500')
        },
        title: {
          color: color('yellow-500')
        }
      }
    }
  }
};

export { alert };
