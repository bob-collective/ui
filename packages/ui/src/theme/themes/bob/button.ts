import { fontSize, fontWeight, lineHeight, rounded, spacing, transition } from '../../core';
import { ButtonTheme } from '../../components';
import { hexToRgba, rem } from '../../utils';

import { color } from './colors';

const button: ButtonTheme = {
  base: {
    borderRadius: rounded('md'),
    fontWeight: fontWeight('normal'),
    ...transition('common', 'normal')
  },
  size: {
    s: {
      height: rem(36),
      fontSize: fontSize('s'),
      lineHeight: lineHeight('s'),
      padding: `0 ${spacing('md')}`
    },
    md: {
      height: spacing('5xl'),
      fontSize: fontSize('s'),
      lineHeight: lineHeight('s'),
      padding: `0 ${spacing('lg')}`
    },
    lg: {
      height: rem(44),
      fontSize: fontSize('md'),
      lineHeight: lineHeight('md'),
      padding: `0 ${spacing('xl')}`
    },
    xl: {
      height: spacing('6xl'),
      fontSize: fontSize('md'),
      lineHeight: lineHeight('md'),
      padding: `0 ${spacing('2xl')}`
    },
    '2xl': {
      height: rem(60),
      fontSize: fontSize('lg'),
      lineHeight: lineHeight('lg'),
      padding: `0 ${spacing('3xl')}`
    }
  },
  active: {
    transform: 'scale(0.97)'
  },
  disabled: {
    opacity: 0.5
  },
  variant: {
    solid: {
      color: {
        default: {
          base: {
            backgroundColor: color('grey-500'),
            color: color('light')
          },
          hover: {
            backgroundColor: color('grey-400')
          }
        },
        primary: {
          base: {
            backgroundColor: color('primary-500'),
            color: color('light')
          },
          hover: {
            backgroundColor: color('primary-600')
          }
        },
        light: {
          base: {
            backgroundColor: color('light'),
            color: color('dark')
          },
          hover: {
            backgroundColor: hexToRgba(color('light'), 80)
          }
        }
      }
    },
    outline: {
      color: {
        default: {
          base: {
            border: `1px solid ${color('grey-300')}`,
            color: color('light')
          },
          hover: {
            backgroundColor: color('grey-300')
          }
        },
        primary: {
          base: {
            border: `1px solid ${color('primary-500')}`,
            color: color('primary-500')
          },
          hover: {
            backgroundColor: color('primary-500'),
            color: color('light')
          }
        },
        light: {
          base: {
            border: `1px solid ${color('light')}`,
            color: color('light')
          },
          hover: {
            backgroundColor: color('light'),
            color: color('dark')
          }
        }
      }
    },
    ghost: {
      color: {
        default: {
          base: {
            color: color('light')
          },
          hover: {
            backgroundColor: hexToRgba(color('grey-300'), 50)
          }
        },
        primary: {
          base: {
            color: color('primary-500')
          },
          hover: {
            backgroundColor: hexToRgba(color('primary-500'), 20)
          }
        },
        light: {
          base: {
            color: color('light')
          },
          hover: {
            backgroundColor: hexToRgba(color('light'), 20)
          }
        }
      }
    }
  }
};

export { button };
