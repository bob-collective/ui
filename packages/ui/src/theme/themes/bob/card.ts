import { CardTheme } from '../../components';
import { transition } from '../../core';

import { color } from './colors';

const card: CardTheme = {
  base: {
    color: color('light'),
    boxShadow: '0px 1px 3px 0px #0000001a, 0px 1px 2px 0px #0000000f',
    backgroundColor: color('grey-400'),
    ...transition('common', 'normal')
  },
  active: {
    transform: 'scale(0.97)'
  },
  disabled: {
    opacity: 0.5
  }
};

export { card };
