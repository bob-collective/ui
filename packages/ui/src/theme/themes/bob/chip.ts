import { ChipTheme } from '../../components';
import { rounded, spacing, transition, typography } from '../../core';

import { color } from './colors';

const chip: ChipTheme = {
  base: {
    color: color('light'),
    backgroundColor: color('grey-400'),
    borderRadius: rounded('full'),
    ...transition('common', 'normal')
  },
  size: {
    s: {
      base: { height: '1.5rem', paddingLeft: spacing('xs'), paddingRight: spacing('xs'), ...typography('xs') },
      content: {
        base: { paddingLeft: spacing('xs'), paddingRight: spacing('xs') },
        adornment: {
          start: { paddingLeft: spacing('xs'), paddingRight: spacing('xs') },
          end: { paddingLeft: spacing('xs'), paddingRight: spacing('xs') },
          both: { paddingLeft: spacing('xs'), paddingRight: spacing('xs') }
        }
      }
    },
    md: {
      base: { height: '1.75rem', paddingLeft: spacing('xs'), paddingRight: spacing('xs'), ...typography('s') },

      content: {
        base: { paddingLeft: spacing('md'), paddingRight: spacing('md') },
        adornment: {
          start: { paddingLeft: spacing('xs'), paddingRight: spacing('md') },
          end: { paddingLeft: spacing('md'), paddingRight: spacing('xs') },
          both: { paddingLeft: spacing('xs'), paddingRight: spacing('xs') }
        }
      }
    },
    lg: {
      base: { height: '2rem', paddingLeft: spacing('md'), paddingRight: spacing('md'), ...typography('md') },
      content: {
        base: { paddingLeft: spacing('md'), paddingRight: spacing('md') },
        adornment: {
          start: { paddingLeft: spacing('xs'), paddingRight: spacing('md') },
          end: { paddingLeft: spacing('md'), paddingRight: spacing('xs') },
          both: { paddingLeft: spacing('xs'), paddingRight: spacing('xs') }
        }
      }
    }
  }
};

export { chip };
