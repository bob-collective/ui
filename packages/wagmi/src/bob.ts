import { ChainId } from '@gobob/chains';
import { defineChain } from 'viem';

const mainnetSourceId = 1; // ethereum

export const bob = defineChain({
  id: ChainId.BOB,
  network: 'BOB',
  name: 'BOB',
  nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
  rpcUrls: {
    default: {
      http: ['https://rpc.gobob.xyz'],
      webSocket: ['wss://rpc.gobob.xyz']
    },
    public: {
      http: ['https://rpc.gobob.xyz'],
      webSocket: ['wss://rpc.gobob.xyz']
    }
  },
  blockExplorers: {
    etherscan: {
      name: 'BOB Explorer',
      url: 'https://explorer.gobob.xyz'
    },
    default: {
      name: 'BOB Explorer',
      url: 'https://explorer.gobob.xyz'
    }
  },
  contracts: {
    multicall3: {
      address: '0xcA11bde05977b3631167028862bE2a173976CA11',
      blockCreated: 23131
    },
    l2OutputOracle: {
      [mainnetSourceId]: {
        address: '0xdDa53E23f8a32640b04D7256e651C1db98dB11C1',
        blockCreated: 4462615
      }
    },
    portal: {
      [mainnetSourceId]: {
        address: '0x8AdeE124447435fE03e3CD24dF3f4cAE32E65a3E',
        blockCreated: 4462615
      }
    }
  }
});

const sepoliaSourceId = 11_155_111; // sepolia

export const oldBobSepolia = defineChain({
  id: ChainId.OLD_BOB_SEPOLIA,
  name: 'BOB Sepolia',
  network: 'bob-sepolia',
  nativeCurrency: {
    decimals: 18,
    name: 'BOB Sepolia',
    symbol: 'ETH'
  },
  rpcUrls: {
    default: {
      http: ['https://testnet.rpc.gobob.xyz']
    },
    public: {
      http: ['https://testnet.rpc.gobob.xyz']
    }
  },
  blockExplorers: {
    default: {
      name: 'BOB Sepolia Explorer',
      url: 'https://testnet-explorer.gobob.xyz'
    }
  },
  contracts: {
    multicall3: {
      address: '0x089b191d95417817389c8eD9075b51a38ca46DE8',
      blockCreated: 2469044
    },
    l2OutputOracle: {
      [sepoliaSourceId]: {
        address: '0x14D0069452b4AE2b250B395b8adAb771E4267d2f',
        blockCreated: 4462615
      }
    },
    portal: {
      [sepoliaSourceId]: {
        address: '0x867B1Aa872b9C8cB5E9F7755feDC45BB24Ad0ae4',
        blockCreated: 4462615
      }
    }
  },
  testnet: true
});

export const bobSepolia = defineChain({
  id: ChainId.BOB_SEPOLIA,
  name: 'BOB Sepolia',
  network: 'bob-sepolia',
  nativeCurrency: {
    decimals: 18,
    name: 'BOB Sepolia',
    symbol: 'ETH'
  },
  rpcUrls: {
    default: {
      http: ['https://bob-sepolia.rpc.gobob.xyz'],
      webSocket: ['wss://bob-sepolia.rpc.gobob.xyz']
    },
    public: {
      http: ['https://bob-sepolia.rpc.gobob.xyz'],
      webSocket: ['wss://bob-sepolia.rpc.gobob.xyz']
    }
  },
  blockExplorers: {
    default: {
      name: 'BOB Sepolia Explorer',
      url: 'https://bob-sepolia.explorer.gobob.xyz'
    }
  },
  contracts: {
    multicall3: {
      address: '0xcA11bde05977b3631167028862bE2a173976CA11',
      blockCreated: 35677
    },
    l2OutputOracle: {
      [sepoliaSourceId]: {
        address: '0x14D0069452b4AE2b250B395b8adAb771E4267d2f',
        blockCreated: 4462615
      }
    },
    portal: {
      [sepoliaSourceId]: {
        address: '0x867B1Aa872b9C8cB5E9F7755feDC45BB24Ad0ae4',
        blockCreated: 4462615
      }
    }
  },
  testnet: true
});
