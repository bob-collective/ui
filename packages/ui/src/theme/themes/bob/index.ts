import { defineTheme } from '../../define';

import { accordion } from './accordion';
import { alert } from './alert';
import { button } from './button';
import { card } from './card';
import { colors } from './colors';
import { dialog } from './dialog';
import { divider } from './divider';
import { drawer } from './drawer';
import { input } from './input';
import { label } from './label';
import { list } from './list';
import { progressBar } from './progress-bar';
import { radio } from './radio';
import { spinner } from './spinner';
import { stepper } from './stepper';
import { _switch } from './switch';
import { table } from './table';
import { tabs } from './tabs';
import { tokenInput } from './token-input';
import { tooltip } from './tooltip';

const bobTheme = defineTheme({
  colors,
  accordion,
  alert,
  button,
  card,
  dialog,
  divider,
  drawer,
  input,
  list,
  radio,
  spinner,
  switch: _switch,
  progressBar,
  tooltip,
  tokenInput,
  table,
  tabs,
  stepper,
  label
});

export { bobTheme };
