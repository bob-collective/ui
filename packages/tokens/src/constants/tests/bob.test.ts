import { describe, it, expect } from 'vitest';

import { bobTokens, bobSepoliaTokens } from '../bob';

describe('Token Export Snapshots', () => {
  it('should match snapshot for bobTokens', () => {
    expect(bobTokens).toMatchSnapshot();
  });

  it('should match snapshot for bobSepoliaTokens', () => {
    expect(bobSepoliaTokens).toMatchSnapshot();
  });
});
