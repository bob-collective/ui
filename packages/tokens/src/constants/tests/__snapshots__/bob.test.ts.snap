// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Token Export Snapshots > should match snapshot for bobSepoliaTokens 1`] = `
{
  "usdc": ERC20Token {
    "address": "0x2c1f73d37E7cdbA2792156139b0d35A765d595C5",
    "chainId": 808813,
    "decimals": 6,
    "isNative": false,
    "isToken": true,
    "name": "Bridged USDC (BOB)",
    "projectLink": undefined,
    "symbol": "USDC",
  },
  "weth": ERC20Token {
    "address": "0x4200000000000000000000000000000000000006",
    "chainId": 808813,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
}
`;

exports[`Token Export Snapshots > should match snapshot for bobTokens 1`] = `
{
  "tbtc": ERC20Token {
    "address": "0xBBa2eF945D523C4e2608C9E1214C2Cc64D4fc2e2",
    "chainId": 60808,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "tBTC v2",
    "projectLink": undefined,
    "symbol": "tBTC",
  },
  "usdc": ERC20Token {
    "address": "0xe75D0fB2C24A55cA1e3F96781a2bCC7bdba058F0",
    "chainId": 60808,
    "decimals": 6,
    "isNative": false,
    "isToken": true,
    "name": "Bridged USDC (BOB)",
    "projectLink": undefined,
    "symbol": "USDC",
  },
  "usdt": ERC20Token {
    "address": "0x05D032ac25d322df992303dCa074EE7392C117b9",
    "chainId": 60808,
    "decimals": 6,
    "isNative": false,
    "isToken": true,
    "name": "Tether USD",
    "projectLink": undefined,
    "symbol": "USDT",
  },
  "wbtc": ERC20Token {
    "address": "0x03C7054BCB39f7b2e5B2c7AcB37583e32D70Cfa3",
    "chainId": 60808,
    "decimals": 8,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped BTC",
    "projectLink": undefined,
    "symbol": "WBTC",
  },
  "weth": ERC20Token {
    "address": "0x0000000000000000000000000000000000000000",
    "chainId": 60808,
    "decimals": 18,
    "isNative": false,
    "isToken": true,
    "name": "Wrapped Ether",
    "projectLink": "https://weth.io",
    "symbol": "WETH",
  },
}
`;
